cmd/node_agent/na/onprem.go

```diff
@@ -39,6 +40,25 @@ func (na *onPremPlatformImpl) IsProperPlatform() bool {
 	return true
 }
 
+func (na *onPremPlatformImpl) GetServiceIdentity(certificateFile string) (string, error) {
+	certBytes, err := ioutil.ReadFile(certificateFile)
+	if err != nil {
+		glog.Errorf("Failed to load cert chain file: %s", err)
+		return "", err
+	}
+	cert, err := pki.ParsePemEncodedCertificate(certBytes)
+	if err != nil {
+		glog.Errorf("Failed to parse cert chain bytes: %s", err)
+		return "", err
+	}
+	serviceIDs := pki.ExtractIDs(cert.Extensions)
+	if len(serviceIDs) != 1 {
+		glog.Errorf("Cert have %v SAN fields, should be 1", len(serviceIDs))
```

Comment:
We should avoid both logging an error and returning en error. Doing this is likely to cause an error being logged more than once.

Reply:
Done